zcash_client_sqlite/src/wallet/init/migrations/addresses_table.rs

```diff
@@ -0,0 +1,176 @@
+
+            // Verify that the address column contains the expected value.
+            let address: String = row.get(2)?;
+            let decoded = RecipientAddress::decode(&self.params, &address).ok_or_else(|| {
+                WalletMigrationError::CorruptedData(format!(
+                    "Could not decode {} as a valid Zcash address.",
+                    address
+                ))
+            })?;
+            let decoded_address = if let RecipientAddress::Unified(ua) = decoded {
+                ua
+            } else {
+                return Err(WalletMigrationError::CorruptedData(
+                    "Address in accounts table was not a Unified Address.".to_string(),
+                ));
+            };
+            let (expected_address, idx) = ufvk.default_address();
+            if decoded_address != expected_address {
+                return Err(WalletMigrationError::CorruptedData(format!(
+                    "Decoded UA {} does not match the UFVK's default address {} at {:?}.",
+                    address,
+                    RecipientAddress::Unified(expected_address).encode(&self.params),
+                    idx,
+                )));
+            }
+
+            // The transparent_address column might not be filled, depending on how this
+            // crate was compiled.
+            if let Some(transparent_address) = row.get::<_, Option<String>>(3)? {
+                let decoded_transparent =
+                    RecipientAddress::decode(&self.params, &transparent_address).ok_or_else(
+                        || {
+                            WalletMigrationError::CorruptedData(format!(
+                                "Could not decode {} as a valid Zcash address.",
+                                address
+                            ))
+                        },
+                    )?;
+                let decoded_transparent_address = if let RecipientAddress::Transparent(addr) =
+                    decoded_transparent
+                {
+                    addr
+                } else {
+                    return Err(WalletMigrationError::CorruptedData(
+                        "Address in transparent_address column of accounts table was not a transparent address.".to_string(),
+                    ));
+                };
+
+                // Verify that the transparent_address column contains the expected value,
+                // so we can confidently delete the column knowing we can regenerate the
+                // values from the stored UFVKs.
+
+                // We can only check if it is the expected transparent address if the
+                // transparent-inputs feature flag is enabled.
+                #[cfg(feature = "transparent-inputs")]
+                {
+                    let expected_address = ufvk
+                        .transparent()
+                        .and_then(|k| k.derive_external_ivk().ok().map(|k| k.default_address().0));
```

Comment:
This is only a valid check because we're assuming that no wallets exist that were initialized after the previous migration (in which case the taddr would be the taddr from the default UA, rather than the one derived at index 0.) I'm not sure that this corruption check is actually useful, but I guess it's okay.